editor/src/components/canvas/canvas-strategies/escape-hatch-strategy.tsx

```diff
@@ -416,3 +405,105 @@ function collectHighlightCommand(
   }, canvasState.selectedElements)
   return showOutlineHighlight('transient', [...siblingFrames, ...draggedFrames])
 }
+
+function findAbsoluteDescendants(
+  selectedElements: Array<ElementPath>,
+  metadata: ElementInstanceMetadataMap,
+): Array<ElementPath> {
+  const isSelectionInsideFocusedComponent = selectedElements.every((path) =>
+    EP.isInsideFocusedComponent(path),
+  )
+  return mapDropNulls((element) => {
+    const elementPath = element.elementPath
+    const isInsideFocusedComponent = EP.isInsideFocusedComponent(elementPath)
+    const allInsideComponentOrOutside =
```

Comment:
I think instead of this you want to make sure that the `path.parts` fully match except for the last part, where they are in an ancestor-descendant relationship